### DISCUSSION TITLE
[WASM][WindowsAppSDK] Updating existing project from .NET6 to .NET7

### DISCUSSION THREAD

**Discussion by DierkDroth** [unoplatform/uno#10414] [Q&A] [answered]
Hi,

I'm working my way through updating my UNO WASM+WindowsAppSDK project from .NET6 to .NET7. A few issues I ran into:

- First thing I noticed: is there a .NET7 equivalent to [this](https://platform.uno/docs/articles/get-started-winui3.html)?
- Next: I get deprecation warnings below. Is that expected?
- To confirm: am I now on the latest and correct UNO templates for .NET7?
- How would I create a fresh, new UNO .NET7 project (as reference)? `dotnet new unoapp-winui-net6 -o MyUnoApp` obviously would not do it...
- when I tried `dotnet new unoapp -o MyUnoApp` would it say something about 'Multi-platform app **.NET6**' being created, which is not what I wanted

I also tried to create a fresh, new UNO project right in VS 17.4 (latest) as per [here](https://platform.uno/docs/articles/getting-started-tutorial-1.html). However, there are no UNO project templates at all in my VS

I'm at a loss a I have no way to create a fresh, new WASM + WindowsAppSDK project for .NET7 :-(

<img width="867" alt="image" src="https://user-images.githubusercontent.com/26179242/201508536-974f5c20-64ee-4216-b80c-a368c458a17f.png">

**Comment by jeromelaban** [comment, marked as answer]
> * First thing I noticed: is there a .NET7 equivalent to [this](https://platform.uno/docs/articles/get-started-winui3.html)?

It's equivalent, net6 and net7 introduce no specific change. You can create a new app without issues [with the visual studio extension](https://platform.uno/docs/articles/get-started-vs-2022.html#install-the-solution-templates), or by using [`dotnet new unoapp`](https://platform.uno/docs/articles/get-started-dotnet-new.html?tabs=windows) and choose .NET 7 as the target framework. This particular document is outdated and will be removed.

> * Next: I get deprecation warnings below. Is that expected?

Just a change in dotnet's behavior, we'll adjust our docs.

> * To confirm: am I now on the latest and correct UNO templates for .NET7?

[Uno 4.6.x supports .NET 7.](https://platform.uno/blog/uno-platform-4-6-hello-net-7/)

> * when I tried `dotnet new unoapp -o MyUnoApp` would it say something about 'Multi-platform app **.NET6**' being created, which is not what I wanted

The `.NET 6` reference is incorrect in the sense that it's indeed a net6 app because you did not specific `-f net7.0`, but [you can follow the docs here to get started with net7.0 and `dotnet new`](https://platform.uno/docs/articles/get-started-dotnet-new.html?tabs=windows).

**Reply by DierkDroth** [reply]
Thanks @jeromelaban. I'll update my documentation links...
